src/openrct2/localisation/Localisation.cpp

```diff
@@ -1228,10 +1228,12 @@ void format_string_to_upper(utf8 *dest, size_t size, rct_string_id format, void
     dest[upperString.size()] = '\0';
 }
 
-money32 string_to_money(char * string_to_monetise)
+money32 string_to_money(const char* string_to_monetise)
 {
+    char processedString[128];
+    safe_strcpy(processedString, string_to_monetise, 128);
```

Comment:
If you're going to allocate a segment of memory for this, you might as well fill it during the initial filter process (`Remove everything except numbers decimal, and minus sign(s)` loop).

This way, you no longer need to copy on each iteration (`Move everything over to the left by one` loop), making the whole process much more efficient.